value-processor/src/org/immutables/value/processor/Immutables.generator

```diff
@@ -1685,13 +1685,47 @@ checkNotIsSet([v.names.isSet](), "[v.names.raw]");
   [else if v.jdkOptional and ((v.jdkSpecializedOptional or v.optionalAcceptNullable) or v.primitiveElement)]
     this.[v.name] = [maybeCopyOf v][v.name][/maybeCopyOf];
   [else if v.jdkOptional]
-    this.[v.name] = [maybeCopyOf v][requireNonNull type]([v.name], "[v.name]")[/maybeCopyOf];
+    this.[v.name] = [if v.isAttributeBuilder][valueToBuilder v.getAttributeBuilderDescriptor][requireNonNull type]([v.name], "[v.name]")[/valueToBuilder];[else][maybeCopyOf v][requireNonNull type]([v.name], "[v.name]")[/maybeCopyOf];[/if]
   [else]
     this.[v.name] = [optionalOf v]([maybeCopyOf v][v.name][/maybeCopyOf]);
   [/if]
     [nondefaultSetInBuilder v]
     return [builderReturnThis type];
   }
+  [if v.isAttributeBuilder]
+
+    /**
+     * Returns a builder for [sourceDocRef v].
+     *
+     * Once called, the attribute builder field is set to a new instance of the builder.
+     * If called more than once, returns the same builder instance.
+     */
+    [eachLine v.builderAttributeAnnotation]
+    [deprecation v]
+    [builderInitAccess v]final [attributeBuilderBuilderType v.getAttributeBuilderDescriptor] [v.names.getBuilder]() {
+      if (this.[v.name] == null) {
+        this.[v.name] = [createAttributeBuilder v.getAttributeBuilderDescriptor];
+      }
+      [nondefaultSetInBuilder v]
```

Comment:
This is directly copied from the non-optional setter flow. I tried to templetize it, but the nondefaultSetInBuilder defined in the let above causes the template to fail. I think you have to pass a function into the template... but can't figure out how to get it to work.